tests/reporter/discord-reporter.js

```diff
@@ -0,0 +1,488 @@
+/* eslint-disable max-lines */
+/* eslint-disable react-func/max-lines-per-function */
+/* eslint-disable no-console */
+
+// Load environment variables from .env file
+require('dotenv').config();
+
+const MAX_RETRIES = process.env.CI ? 2 : 0; // 2 retries on CI, 0 retry locally
+
+const botTokenEnv = process.env.DISCORD_BOT_TOKEN;
+const channelIdEnv = process.env.DISCORD_CHANNEL_ID; // Channel ID to send messages to
+const notificationRoleIdEnv = process.env.DISCORD_NOTIFICATION_ROLE_ID; // Optional role to ping on failures
```

Comment:
Please add these to `.env.example` and also in the body of the PR to be visible for the person responsible for deployments.